PT-2134 - support for PostgreSQL and MySQL operators

svetasmirnova opened the pull request:
Support for all operators in pt-k8s-debug-collector.

Here is what changed:

- Option --resource now supports pxc, ps, psmdb, psql, and none. Default is none
- Two new options added: --kubeconfig and --forwardport. These allow to specify custom configuration file for kubectl and custom port when collecting summary from the database
- Test case now uses --kubeconfig, --forwardport options, and environment variables KUBECONFIG_PXC, KUBECONFIG_PS, KUBECONFIG_PSMDB, KUBECONFIG_PSQL, FORWARDPORT which allow more flexibility.
- Output file pt-summary.txt renamed to summary.txt to avoid confusion with the pt-summary tool 
- Summaries now collected for all Percona operators unless --resources=none is specified. For PXC, MySQL, and MongoDB corresponding summary tool (pt-mysql-summary and pt-mongodb-summary) must be in PATH.
- README.md was replaced with link to README.rst to avoid duplicate effort while working on the tool's documentation

svetasmirnova committed:
Changelog and version for Release 3.5.0

EvgeniyPatlan committed:
PT-2076 add rhel9 dependency

EvgeniyPatlan committed:
Merge pull request #561 from EvgeniyPatlan/release-3.5.0

PT-2076 add rhel9 dependency

percona-csalguero committed:
PT-2128 Updated version to 3.5.0

percona-csalguero committed:
Merge branch 'PT-2128_Set_version_3.5' into release-3.5.0

svetasmirnova committed:
PT-2134 - Support for PostgreSQL and PS MySQL operators

For PS MySQL operator: added support of pt-mysql-summary
For PostgreSQL operator: added support for pg_gather
Options added:
  - kubeconfig - path to kubeconfig
  - forwardport - port to use when calling pt-*-summary tools ad pt_gather
Options changed:
  - resource - now default value is  (was pxc). New values added:
    psql - for PostgreSQL operator
    ps - for PS MySQL operator
    if default value () is used: only K8 information is collected
    Otherwise resource-specific summary is collected

svetasmirnova committed:
PT-2134_support_for_PostgreSQL_and_MySQL_operators

Fixed summary collection when connecting to PostgreSQL opertor and no resource specified

svetasmirnova committed:
PT-2134 - support for PostgreSQL and MySQL operators

Added test case for full supported set of values for option --resource,
added --kubeconfig and --forwardport options for test cases,
added support for environment variables KUBECONFIG_PXC, KUBECONFIG_PS, KUBECONFIG_PSMDB, KUBECONFIG_PSQL, FORWARDPORT for test cases

svetasmirnova committed:
PT-2134 - support for PostgreSQL and MySQL operators

updated docs/pt-k8s-debug-collector.rst,
replaced README.md in src/go/pt-k8s-debug-collector wih symbolic link to docs/pt-k8s-debug-collector.rst
to avoid having inconsistent documentation

svetasmirnova committed:
PT-2134 - support for PostgreSQL and MySQL operators

Removed curl STDERR from the command output

svetasmirnova committed:
PT-2134 - support for PostgreSQL and MySQL operators

typo in pt-k8s-debug-collector.rst

svetasmirnova committed:
Merge branch '3.x' into PT-2134_support_for_PostgreSQL_and_MySQL_operators

egegunes commented on src/go/pt-k8s-debug-collector/dumper/dumper.go:
```diff
@@ -78,16 +77,32 @@ func New(location, namespace, resource string) Dumper {
 				"perconaservermongodbrestores",
 				"perconaservermongodbs",
 			)
+		} else if resourceType(resource) == "psql" {
```
I'd call it `pg` but it's up to you

svetasmirnova replied:
Good idea, Support team also likes it. So made the change.

svetasmirnova committed:
PT-2134 - support for PostgreSQL and MySQL operators

Renamed --resource=psql to --resource=pg after review suggestion by Ege
and collecting feedback from potential users

svetasmirnova committed:
Merge branch 'PT-2134_support_for_PostgreSQL_and_MySQL_operators' of github.com:percona/percona-toolkit into PT-2134_support_for_PostgreSQL_and_MySQL_operators
